fundrebalancer/fundrebalancer.go

```diff
@@ -244,21 +244,45 @@ func (r *FundRebalancer) MoveFundsToChain(
 			return nil, nil, fmt.Errorf("getting txns required for fund rebalancing: %w", err)
 		}
 
+		chainConfig, err := config.GetConfigReader(ctx).GetChainConfig(rebalanceFromChainID)
+		if err != nil {
+			return nil, nil, fmt.Errorf("getting chain config for gas threshold check: %w", err)
+		}
+
+		if chainConfig.MaxRebalancingGasThreshold != 0 {
+			// Check if total gas needed exceeds threshold to rebalance funds from this chain
+			totalRebalancingGas, err := r.estimateTotalGas(txns)
+			if err != nil {
+				return nil, nil, fmt.Errorf("estimating total gas for transactions: %w", err)
+			}
+
+			if totalRebalancingGas > chainConfig.MaxRebalancingGasThreshold {
+				lmt.Logger(ctx).Info(
+					"skipping rebalance from chain "+rebalanceFromChainID+" due to rebalancing txs exceeding gas threshold",
+					zap.String("sourceChainID", rebalanceFromChainID),
+					zap.String("destinationChainID", rebalanceToChain),
+					zap.Uint64("estimatedGas", totalRebalancingGas),
+					zap.Uint64("gasThreshold", chainConfig.MaxRebalancingGasThreshold),
+				)
+				continue
+			}
+		}
+
```

Comment:
I feel like this should be moved to a helper function, maybe something like
```go
func (r *FundRebalancer) isGasAcceptable(...) {
  // this logic
}
```

Reply:
done